core-models/src/main/scala/com/mesosphere/usi/core/models/RunSpec.scala

```diff
@@ -2,6 +2,5 @@ package com.mesosphere.usi.core.models
 
 /**
   * Specification used to launch a [[PodSpec]]
-  * @param cpus number of cpus requested by the pod
   */
-case class RunSpec(cpus: Double)
+case class RunSpec(resourceRequirements: Seq[ResourceRequirement], commandBuilder: CommandInfoGenerator)
```

Comment:
I don't know how I feel about `commandBuilder` in the `RunSpec`. I guess it's here because we wanted to keep it flexible for the orchestrators? But on the other hand, orchestrators should be primarily about business logic and not about building Mesos protobufs correctly. Maybe we could _make it extendable later_ similar to how plugins in Marathon _can but not must_ override `TaskInfo` and `TaskGroupInfo`?

Reply:
We already discussed on slack but just for reference. The usage of this is  `RunSpec(SimpleShellCommandInfoGenerator("echo 'abc'"))`. I would personally prefer something like `RunSpec(ShellCommand("cmd"))` but we can iterate on that

Reply:
Yeah, I'm not exactly sure at the moment. 

https://jira.mesosphere.com/browse/DCOS-48503
https://jira.mesosphere.com/browse/DCOS-47481

Reply:
Replaced with string shell command for now, with call out to above JIRAs